core/store/src/db/rocksdb/instance_tracker.rs

```diff
@@ -0,0 +1,179 @@
+use once_cell::sync::Lazy;
+use std::sync::{Condvar, Mutex};
+use tracing::info;
+
+/// Describes number of RocksDB instances and sum of their max_open_files.
+///
+/// This is modified by [`InstanceTracker`] which uses RAII to automatically
+/// update the trackers when new RocksDB instances are created and dropped.
+#[derive(Default)]
+struct Inner {
+    /// Number of RocksDB instances open.
+    count: usize,
```

Comment:
ultra minor, but I'd use `u32/u64` here. We are not indexing, so theer's no need to use usize.

Reply:
But what if we need five billion instances? O_o

Reply:
On more serious note, for me `usize` is just general integer type.  If I’m to use sized integer I feel I need a reason for the specified number of bits.